pkg/virt-handler/isolation/isolation.go

```diff
@@ -164,43 +168,81 @@ func (s *socketBasedIsolationDetector) AdjustResources(vm *v1.VirtualMachineInst
 		return nil
 	}
 
-	// bump memlock ulimit for libvirtd
-	res, err := s.Detect(vm)
+	isolationResult, err := s.Detect(vm)
+	if err != nil {
+		return err
+	}
+
+	// make the best estimate for memory required by libvirt
+	memlockSize, err := getMemlockSize(vm)
 	if err != nil {
 		return err
 	}
-	launcherPid := res.Pid()
 
 	processes, err := ps.Processes()
 	if err != nil {
 		return fmt.Errorf("failed to get all processes: %v", err)
 	}
 
-	for _, process := range processes {
-		// consider all processes that are virt-launcher children
-		if process.PPid() != launcherPid {
-			continue
+	// bump libvirtd and qemu processes locked memory limits
+	libvirtProcesses := filterVirtLaunchersLibvirtProcesses(isolationResult, processes)
+	for _, process := range libvirtProcesses {
+		err = setProcessMemoryLockLimit(process.Pid(), uint64(memlockSize), uint64(memlockSize))
+		if err != nil {
+			return fmt.Errorf("failed to set rlimit for memory lock: %v", err)
 		}
+	}
 
-		// libvirtd process sets the memory lock limit before fork/exec-ing into qemu
-		if process.Executable() != "libvirtd" {
-			continue
-		}
+	return nil
+}
 
-		// make the best estimate for memory required by libvirt
-		memlockSize, err := getMemlockSize(vm)
-		if err != nil {
-			return err
+func filterVirtLaunchersLibvirtProcesses(isolationResult IsolationResult, processes []ps.Process) []ps.Process {
+	var filteredProcesses []ps.Process
+
+	// consider all processes that are virt-launcher children
+	launcherPid := isolationResult.Pid()
+
+	// consider virt-launcher root process id which is QEMU process parent process
+	launcherParentPid := isolationResult.PPid()
+
+	filteredProcesses = filterProcessesByParentPID(processes, []int{launcherPid, launcherParentPid})
+	// libvirtd process sets the memory lock limit before fork/exec-ing into qemu
+	// Its necessary to change QMEU process limits if its already exists (e.g: post migration just before hostdev)
```

Comment:
What if we are in a reconcile? Are you sure we can reach this only on post-migration?

Reply:
We will reach this also from reconcile.
Do we want to restrict changing QEMU process limits only on post migration?
If so I can create `AdjsutResourcesOnPostMigration` function 
that will be called on `finalizeMigration` (instead of regular `AdjustResources`)
@EdDev What you think?

Reply:
I think it is less risky if the change is done per the specific need known to us. The qemu process memlock limit needs to be changed only when the VMI has a VFIO device and the state is post-migration.
If this is open to other scenarios, we may get into corner cases where we should not be.

> If so I can create AdjsutResourcesOnPostMigration function

As I see it, a function name should express what it does, not who or when it is called.

Reply:
I pushed new changes so that only QEMU process will be adjusted on post migration